Automattic/jetpack
# Support Google Analytics tracking for WooCommerce events on AMP

This PR adds support for Google Analytics tracking for WooCommerce events on AMP. It currently supports the two "basic" events (Add to Cart, and Purchase transactions), but should be able to be extended to support the six "Enhanced eCommerce" events in the future. 


* Support Google Analytics tracking for WooCommerce events on AMP

#### Testing instructions:






* Install AMP plugin and set it to Transitional mode. 
* Install WooCommerce plugin and set up a basic store with at least one product
* Set up a Google Analytics account and configure a test property
* In Jetpack Settings -> Traffic -> Google Analytics, click "Configure your Google Analytics settings"
  * On this page, scroll to the Google Analytics section and enable tracking, provide the GA tracking ID for the property you set up, and enable the two "Basic store analytics" checkboxes
  * Save settings
* On the front end of the site, go to the product you added and enable AMP mode (`?amp`)
* Click the Add to Cart button, wait for the XHR to complete, and reload/navigate the page
* In the inspector, search for "collect" to filter the events that are sent to GA and look at the query string params for the requests. One will be the pageview itself, and one will be for the Add to Cart event and will include information about the product you added. 
* Proceed to the checkout page and complete your checkout
* In the inspector, search for "collect" to filter the events that are sent to GA and look at the query string params for the requests. One will be the pageview itself, and one will be for the Purchase event and will include information about the product you added. 
* Wait ~10 minutes for GA to sync up with the data
* Go to GA and in the left nav, select Conversions -> Ecommerce -> Product List Performance. Here you should see numbers in the "Product Adds to Cart" and "Product Checkout" columns that reflect the above actions. 

#### Proposed changelog entry for your changes:

* Add support for basic Google Analytics tracking for WooCommerce events on AMP
